src/ResourceManager/Resources/Commands.ResourceManager/Cmdlets/Components/LongRunningOperationHelper.cs

```diff
@@ -101,6 +101,43 @@ internal string WaitOnOperation(OperationResult operationResult)
 
             this.ProgressTrackerObject.UpdateProgress("Complete", 100);
 
+            // Handle async operation status
+            var result = JObject.Parse(operationResult.Value);
+
+            JToken statusToken;
+            if (result.TryGetValue("status", out statusToken))
+            {
+                TerminalProvisioningStates resourceProvisioningState;
+                if (Enum.TryParse(value: statusToken.ToString(), ignoreCase: true, result: out resourceProvisioningState))
+                {
+                    if (resourceProvisioningState == TerminalProvisioningStates.Succeeded ||
+                    resourceProvisioningState == TerminalProvisioningStates.Ready)
+                    {
+                        return this.GetResourcesClient()
+                            .GetResource<JObject>(resourceId, apiVersion, CancellationToken.None)
+                            .Result
+                            .ToString();
+                    } else if (resourceProvisioningState == TerminalProvisioningStates.Failed ||
+                        resourceProvisioningState == TerminalProvisioningStates.Canceled)
+                    {
+                        var error = JObject.Parse(operationResult.Value);
```

Comment:
You are doing this on line 105 too. why do it again?

Reply:
Fixed